ios/Classes/VideoCompressPlugin.m

```diff
@@ -1,8 +1,100 @@
 #import "VideoCompressPlugin.h"
-#import <video_compress/video_compress-Swift.h>
+#import "Utility.h"
+#import <AVFoundation/AVFoundation.h>
+
+@interface VideoCompressPlugin ()
+
+- (void)compressVideo:(NSString *)path quality:(NSNumber *)quality result:(FlutterResult)result;
+- (void)cancelCompression:(FlutterResult)result;
+- (NSDictionary *)getMediaInfoJson:(NSString *)path;
+- (NSString *)getExportPreset:(NSNumber *)quality;
+
+@end
+
+@implementation VideoCompressPlugin {
+    AVAssetExportSession *exporter;
+    bool stopCommand;
+}
 
-@implementation VideoCompressPlugin
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
-  [SwiftVideoCompressPlugin registerWithRegistrar:registrar];
+    FlutterMethodChannel *channel =
+    [FlutterMethodChannel methodChannelWithName: @"video_compress"
+                                binaryMessenger:[registrar messenger]];
+    VideoCompressPlugin *instance = [VideoCompressPlugin new];
+    [registrar addMethodCallDelegate:instance channel:channel];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+    NSDictionary *args = call.arguments;
+    if ([@"compressVideo" isEqualToString:call.method]) {
+        NSString *path = args[@"path"];
+        NSNumber *quality = args[@"quality"];
+        [self compressVideo:path quality:quality result:result];
+    } else if ([@"cancelCompression" isEqualToString:call.method]) {
+        [self cancelCompression:result];
+    } else if ([@"deleteAllCache" isEqualToString:call.method]) {
+        [Utility deleteFile:[Utility basePath]];
+        result(@YES);
+    } else {
+        result(FlutterMethodNotImplemented);
+    }
+}
+
+- (NSDictionary *)getMediaInfoJson:(NSString *)path {
+    NSDictionary* dictionary = @{ @"path": [Utility excludeFileProtocol:path] };
+    return dictionary;
+}
+
+- (NSString *)getExportPreset:(NSNumber *)quality {
+    switch ([quality intValue]) {
+        case 1:
+            return AVAssetExportPresetLowQuality;
+        case 2:
+            return AVAssetExportPresetMediumQuality;
+        case 3:
+            return AVAssetExportPresetHighestQuality;
+        default:
+            return AVAssetExportPresetMediumQuality;
+    }
+}
+
+- (void)compressVideo:(NSString *)path quality:(NSNumber *)quality result:(FlutterResult)result {
+    NSURL *sourceVideoUrl = [Utility getPathUrl:path];
+    AVURLAsset *sourceVideoAsset = [AVURLAsset URLAssetWithURL:sourceVideoUrl options:nil];
+    
+    NSURL *compressionUrl = [Utility getPathUrl:[NSString stringWithFormat:@"%@/%@.mp4", [Utility basePath], [Utility getFileName:path]]];
+    
+    stopCommand = false;
+    exporter = [AVAssetExportSession exportSessionWithAsset:sourceVideoAsset presetName: [self getExportPreset:quality]];
```

Comment:
if we call compressVideo twice old exporter object will be deallocated

Reply:
There's a guard on flutter side: https://github.com/jonataslaw/VideoCompress/blob/master/lib/src/video_compressor.dart#L135

Reply:
its in different class, its easy to miss this scenario if someone refactor and dont have unit tests

Reply:
What's your suggestion?

Reply:
before initialise cancel previous operation

Reply:
It will be complicated because if I call cancel, I will need to wait for the `exportAsynchronouslyWithCompletionHandler` of the canceled operation to run before starting the new compression. (Perhaps by queueing the operation and calling it from `exportAsynchronouslyWithCompletionHandler`)

It can be done, but I feel that it's unneeded complexity, unless there's an easier way to do it.